modules/nf-core/snpsift/dbnsfp/main.nf

```diff
@@ -0,0 +1,47 @@
+process SNPSIFT_DBNSFP {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::snpsift=5.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/snpsift:5.1d--hdfd78af_0' :
+        'quay.io/biocontainers/snpsift:5.1d--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(vcf), path(vcf_tbi)
+    tuple val(meta2), path(database), path(dbs_tbi)// TBI files are optional (use when compressed VCF file)
+
+    output:
+    tuple val(meta), path("*.vcf"), emit: vcf
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def algorithms = task.ext.algorithms ? "-f ${task.ext.algorithms}": ''
```

Comment:
Can't algorithms be specified through ext.args?

Reply:
Yes they can be specified in ext.args The amount of algorithms that can be used is big. Plus the other parameters. The introduction of this variable is to help in the customization of algorithms to be used and use ext.args for the other optional parameters.

Reply:
If only one external argument is used, apart from visibility, usage may be misunderstood:
ext.args = [
    --arg1 blabla
    --arg2 blabla
    --arg3 blabla
    --arg4 blabla
    --arg5 blabla
    --arg6 blabla
    --arg7 blabla
    -f Uniprot_acc, Interpro_domain, SIFT_pred, Polyphen2_HDIV_pred, \
    Polyphen2_HVAR_pred, LRT_pred, MutationTaster_pred, GERP++_NR, \
    GERP++_RS, phastCons100way_vertebrate, 1000Gp1_AF, 1000Gp1_AFR_AF, \
    1000Gp1_EUR_AF, 1000Gp1_AMR_AF, 1000Gp1_ASN_AF, ESP6500_AA_AF, ESP6500_EA_AF, \
    MutationTaster_pred, MutationAssessor_pred, FATHMM_pred, PROVEAN_pred, CADD_phred, \
    MetaSVM_pred, 1000Gp3_AC, 1000Gp3_AF, 1000Gp3_AFR_AC, 1000Gp3_AFR_AF, 1000Gp3_EUR_AC, \
    1000Gp3_EUR_AF, 1000Gp3_AMR_AC, 1000Gp3_AMR_AF, 1000Gp3_EAS_AC, 1000Gp3_EAS_AF, \
    1000Gp3_SAS_AC, 1000Gp3_SAS_AF, ESP6500_AA_AC, ESP6500_AA_AF, ESP6500_EA_AC, \
    ESP6500_EA_AF, ExAC_AC, ExAC_AF, ExAC_Adj_AC, ExAC_Adj_AF, ExAC_AFR_AC, ExAC_AFR_AF, \
    ExAC_AMR_AC, ExAC_AMR_AF, ExAC_EAS_AC, ExAC_EAS_AF, ExAC_FIN_AC, ExAC_FIN_AF, ExAC_NFE_AC, \
    ExAC_NFE_AF, ExAC_SAS_AC, ExAC_SAS_AF
]

If two external arguments are used (args & algorithms), the usage of the program is more clear as you can better manipulate the arguments according the algorithms used,

ext.args = [
    --arg1
    --arg2
    --arg3
    --arg4
    --arg5
    --arg6
    --arg7
]
ext.algorithms = [
    Uniprot_acc, Interpro_domain, SIFT_pred, Polyphen2_HDIV_pred, \
    Polyphen2_HVAR_pred, LRT_pred, MutationTaster_pred, GERP++_NR, \
    GERP++_RS, phastCons100way_vertebrate, 1000Gp1_AF, 1000Gp1_AFR_AF, \
    1000Gp1_EUR_AF, 1000Gp1_AMR_AF, 1000Gp1_ASN_AF, ESP6500_AA_AF, ESP6500_EA_AF, \
    MutationTaster_pred, MutationAssessor_pred, FATHMM_pred, PROVEAN_pred, CADD_phred, \
    MetaSVM_pred, 1000Gp3_AC, 1000Gp3_AF, 1000Gp3_AFR_AC, 1000Gp3_AFR_AF, 1000Gp3_EUR_AC, \
    1000Gp3_EUR_AF, 1000Gp3_AMR_AC, 1000Gp3_AMR_AF, 1000Gp3_EAS_AC, 1000Gp3_EAS_AF, \
    1000Gp3_SAS_AC, 1000Gp3_SAS_AF, ESP6500_AA_AC, ESP6500_AA_AF, ESP6500_EA_AC, \
    ESP6500_EA_AF, ExAC_AC, ExAC_AF, ExAC_Adj_AC, ExAC_Adj_AF, ExAC_AFR_AC, ExAC_AFR_AF, \
    ExAC_AMR_AC, ExAC_AMR_AF, ExAC_EAS_AC, ExAC_EAS_AF, ExAC_FIN_AC, ExAC_FIN_AF, ExAC_NFE_AC, \
    ExAC_NFE_AF, ExAC_SAS_AC, ExAC_SAS_AF
]

Reply:
Hmmm.. I understand what you are saying, but I think it might do the opposite of what you want. 
nf-core community in general has been moving towards ext.args as the default way to supply non-file arguments. Check pt 3 in https://nf-co.re/docs/contributing/modules#general. nf-core pipeline developers and users know that while configuring modules, non-file arguments have to be supplied through ext.args and so using custom ones will simply throw them off. Thanks to closures and lists, users already have plenty of flexibility to write and break down the arguments as they see fit while passing it into ext.args. So I don't think this specific case warrants its own custom argument.

Reply:
alright, now it should be good